### ISSUE TITLE
knife-windows does not respect proxy exceptions

### ISSUE THREAD

**Issue by smurawski** [chef/knife-windows#321]
Both ENV['NO_PROXY'] and native proxy configuration exceptions are ignored when using knife winrm and knife bootstrap windows.

this means that all attempts to bootstrap are forced to go through a proxy, even when they should not.

**Comment by btm** [comment]
Do you mean during bootstrap? https://github.com/chef/knife-windows/pull/58 added a `--bootstrap-no-proxy` command, but I wonder if we need to set NO_PROXY in `lib\chef\knife\bootstrap\windows-chef-client-msi.erb` similar to how we set HTTP_PROXY there.

What do you mean by native proxy configuration? The "internet explorer" (of old) settings? Do those settings not get exposed as environment variables by Windows?

**Comment by mwrock** [comment]
@btm thats right (re windows ignoring the env variables). mixlib-install does this the right way by setting the proxy on the .net based web client. When we move knife-windows to core, we might consider using mixlib-install for much of the bootstrap logic.

**Comment by btm** [comment]
Looking at https://github.com/chef/knife-windows/pull/362, maybe we need to set `BypassList` in wget.ps1 [too]?
